src/pages/api/auth-connect/verify-validation-code.ts

```diff
@@ -41,7 +42,12 @@ const handler: Handler = async (request: ExtendedRequest, response) => {
             value,
           };
 
-          await addIdentityToUser(body);
+          const newIdentity = await addIdentityToUser(body);
+
+          if (primary) {
+            const { id: identityId } = newIdentity?.data?.addIdentityToUser;
+            await markIdentityAsPrimary(identityId);
+          }
```

Comment:
What is if `newIdentity` is undefined (i.e. `addIdentityToUser` fails)? Same with `markIdentityAsPrimary`.

Reply:
Since there wasn't any error handling on `addIdentityToUser` before my changes, I assumed that was part of the new errors handling plan you're working on. So I didn't had any checks for that. Was I wrong ?

Reply:
What I'm working on concerns `withAuth` middleware :).
We don't handle errors from `request.session.get` and `getDBUserFromSub` in this file because they are done in `withAuth`, and we haven't found an elegant way to reflect it in the handler yet.

`addIdentityToUser` returns `{data, errors}`. Both can be undefined. You need to react accordingly to the truth of these two objects. Same goes for `markIdentityAsPrimary` :). Let me know if it's unclear.

That's also why we try to avoid using `?` when accessing object's keys, because it basically tells "don't handle the exception if one is thrown".

Reply:
Alright, I think I get it. Working on it ;)